### ISSUE TITLE
Editor removes custom config of first shown entity

### ISSUE THREAD

**Issue by philrich** [home-assistant/frontend#19327]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Since some days I have a problem with the Home Assistant Editor. Whenever I edit a Lovelace card, the editor removes all custom config of the first shown entity. Home Assistant is installed via HAOS on a Raspi 4. This setup was workung flawlessly in the past. I have a second Home Assistant Instance running on a Raspi Zero W which does not show this behaviour. Both Instances are running on the latest Home Assistant Version (2024.1.2). I also tried with Safe Mode but then nearly none of my Dashboards are shown, because I make extensive use of card_mod and layout-card.

This is an example card showing the removed card_mod modifications when calling the editor:
![image](https://github.com/home-assistant/frontend/assets/17569738/180623bc-09e7-464f-8bda-043db3289f3f)

This is nearly the same card on my second HA Instance, where everything works as expected:
![image](https://github.com/home-assistant/frontend/assets/17569738/b2e57a36-b3e1-4087-a60f-4225766ba15a)

I removed all custom Integrations and Frontend Components I don't need but it didn't fix the Problem. As of now I am using: card-tools, card-mod, auto-entities, layout-card, mushroom, horizon card, kiosk mode, vertical-stack-in-card, apexcharts-card, energy-overview-card, comfortable-environment-card

### Describe the behavior you expected

The frontend editor should not remove custom config of cards

### Steps to reproduce the issue

Editor removes custom config of first entity on every card it is called on

### What version of Home Assistant Core has the issue?

2024.1.2

### In which browser are you experiencing the issue with?

Google Chrome Version 120.0.6099.199 (Official Build) (64-bit) [also tested on iOS Companion App and Firefox 115.6.0esr]

### Which operating system are you using to run this browser?

Debian 12

**Comment by karwosts** [comment]
Is this consistently reproducible, and if so, can you provide clear steps to reproduce it, including exact yaml of the entire stack that shows this?

**Comment by ildar170975** [comment]
Not a Frontend issue.
https://github.com/thomasloven/lovelace-card-mod/issues/337

**Comment by robinshlee** [comment]
This is my observation.  I had added a card_mod style that made that background of the card blue, saved it and it displayed the card with the blue background as expected.  However, when I went into the editor, the blue background disappeared and switching to the YAML editor mode showed that the custom codes had vanished.  When I aborted the editor (cancel and not save), the card once again displayed the blue background.  So I think the mods were saved earlier and was effective, but will be wiped clean if you try to edit them again.
